crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -1780,3 +1833,19 @@ impl rpce::HeaderlessEncode for GetReaderIconReturn {
         + self.data.len() // dst.write_slice(&self.data);
     }
 }
+
+fn expect_charset(charset: Option<CharacterSet>) -> PduResult<CharacterSet> {
+    if charset.is_none() {
+        return Err(other_err!("internal error: missing character set"));
+    }
+    Ok(charset.unwrap())
```

Comment:
suggestion: It’s more idiomatic (and `unwrap`-free) like this:
```suggestion
    if let Some(charset) = charset {
        Ok(charset)
    } else {
        Err(other_err!("internal error: missing character set"))
    }
```